src/services/codefixes/importFixes.ts

```diff
@@ -717,13 +720,13 @@ namespace ts.codefix {
             readonly name: string;
         };
     }
-    function addNewImports(changes: textChanges.ChangeTracker, sourceFile: SourceFile, moduleSpecifier: string, quotePreference: QuotePreference, imports: ImportsCollection, blankLineBetween: boolean): void {
+    function getNewImports(moduleSpecifier: string, quotePreference: QuotePreference, imports: ImportsCollection): Statement[] {
         const quotedModuleSpecifier = makeStringLiteral(moduleSpecifier, quotePreference);
+        const statements = [];

```

Comment:
if its rare to add multiple imports for same specifier, would it be better to return Statement | Statement[] instead so as to avoid having to allocate arrays for each module specifier if its not common?

Reply:
I also find it confusing that the parameters accommodate this pattern but the arguments do not.

Reply:
I think I've mentioned this before, but I'm kind of surprised that we don't have a type and a collection of helper functions for manipulating `T | T[]`.